starlette/routing.py

```diff
@@ -121,13 +124,17 @@ def compile_path(
     idx = 0
     param_convertors = {}
     for match in PARAM_REGEX.finditer(path):
-        param_name, convertor_type = match.groups("str")
+        param_name, convertor_type, args = match.groups("str")
         convertor_type = convertor_type.lstrip(":")
         assert (
             convertor_type in CONVERTOR_TYPES
         ), f"Unknown path convertor '{convertor_type}'"
-        convertor = CONVERTOR_TYPES[convertor_type]
-
+        if args:  # pragma: no cover
```

Comment:
Hi @Mng-dev-ai,

instead of `pragma: no cover` the `if-else`, you could add a separate test for the `compile_path` function altogether, since it is actually a key compoment of routing.

Reply:
Hi @bibajz,
I'm not sure if it should be added to testing or not.
what do you think @Kludex ?

Reply:
Hello, it seems to me that you misunderstood the meaning of `# pragma: no cover`. It is usually found during conditional imports or interface declaration, not within a core functionality. You can try to grep for its usages -

```
$ grep -rHn --after-context=3 --before-context=3 'pragma: \w*\W*cover$' starlette/
...
```

and as an example, see here:

```
starlette/requests.py-12-try:
starlette/requests.py-13-    from multipart.multipart import parse_options_header
starlette/requests.py:14:except ImportError:  # pragma: nocover
starlette/requests.py-15-    parse_options_header = None
```

As you can see, a conditional import.

Therefore I see it as a misuse of this `pragma` and a sign that tests specifically for `compile_path` need to be added.
Also, your previous implementation was incorrect, which also adds to the necessity of tests.

Reply:
Well... You already added a test that covers this, haven't you?

Reply:
@bibajz Yes, I think i misunderstood the meaning of #  pragma: no cover
@Kludex I meant for the compile path part which seems I have to add a test for it too as @bibajz declared above